pkg/daemon/criruntime/factory.go

```diff
@@ -200,16 +200,39 @@ func detectRuntime(varRunPath string) (cfgs []runtimeConfig) {
 	{
 		_, err1 := os.Stat(fmt.Sprintf("%s/docker.sock", varRunPath))
 		_, err2 := os.Stat(fmt.Sprintf("%s/dockershim.sock", varRunPath))
-		if err1 == nil && err2 == nil {
+		_, err3 := os.Stat(fmt.Sprintf("%s/cri-dockerd.sock", varRunPath))
+		var (
+			runtimeURI       string
+			runtimeRemoteURI string
+		)
+		switch {
+		case err1 == nil && err2 == nil && err3 == nil:
+			klog.Infof("Found multiple docker remote runtime sockets, use %s/cri-dockerd.sock", varRunPath)
+			runtimeURI = fmt.Sprintf("unix://%s/docker.sock", varRunPath)
+			runtimeRemoteURI = fmt.Sprintf("unix://%s/cri-dockerd.sock", varRunPath)
+		case err1 == nil && err2 == nil:
+			runtimeURI = fmt.Sprintf("unix://%s/docker.sock", varRunPath)
+			runtimeRemoteURI = fmt.Sprintf("unix://%s/dockershim.sock", varRunPath)
+		case err1 == nil && err3 == nil:
+			runtimeURI = fmt.Sprintf("unix://%s/docker.sock", varRunPath)
+			runtimeRemoteURI = fmt.Sprintf("unix://%s/cri-dockerd.sock", varRunPath)
+		case err1 == nil && err2 != nil:
+			klog.Errorf("%s/docker.sock exists, but not found %s/dockershim.sock", varRunPath, varRunPath)
+		case err1 != nil && err2 == nil:
+			klog.Errorf("%s/dockershim.sock exists, but not found %s/docker.sock", varRunPath, varRunPath)
+		case err1 == nil && err3 != nil:
+			klog.Errorf("%s/docker.sock exists, but not found %s/cri-dockerd.sock", varRunPath, varRunPath)
+		case err1 != nil && err3 == nil:
+			klog.Errorf("%s/cri-dockerd.sock exists, but not found %s/docker.sock", varRunPath, varRunPath)
+		default:
+			klog.Errorf("Failed to stat docker runtime sockets: %v, %v, %v", err1, err2, err3)
+		}
+		if len(runtimeURI) > 0 && len(runtimeRemoteURI) > 0 {
 			cfgs = append(cfgs, runtimeConfig{
 				runtimeType:      ContainerRuntimeDocker,
-				runtimeURI:       fmt.Sprintf("unix://%s/docker.sock", varRunPath),
```

Comment:
is it possible to treat cri-docker as if it is ContainerRuntimeCommonCRI? you can refer to the cri-o implementation

Reply:
done

Reply:
plz make the code a seperate block, e.g. 

```
	// cri-dockerd
	{
		if _, err = os.Stat(fmt.Sprintf("%s/cri-dockerd.sock", varRunPath)); err == nil {
			cfgs = append(cfgs, runtimeConfig{
				runtimeType:      ContainerRuntimeCommonCRI,
				runtimeRemoteURI: fmt.Sprintf("unix://%s/cri-dockerd.sock", varRunPath),
			})
		}
	}
```

Reply:
done again ~